ansible/roles/dev-desktop/templates/clean-unused-checkouts.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+#
+# {{ ansible_managed }}
+#
+
+# Clean up unused checkouts
+#
+# This script is used to find old checkouts that are no longer in use. Given the
+# size of the build directory, regularly cleaning them up can save significant
+# amounts of disk space.
+
+# Enable strict mode for Bash
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+# Print directories and their size instead of deleting them
+dry_run=false
+
+# Default to search for checkouts older than 60 days
+time="60"
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --dry-run)
+      dry_run=true
+      shift # past argument
+      ;;
+    -t|--time)
+      time="${2}"
+      shift # past argument
+      shift # past value
+      ;;
+    -*)
+      echo "Unknown option $1"
+      exit 1
+      ;;
+  esac
+done
+
+# Find all build or target directories created by users
+#
+# This command combines (`-o`) two different conditions to find all build and
+# target directories that users have created. Within each home directory, we
+# recursively look for directories that either have a file named `x.py` and a
+# directory named `build`, or a file named `Cargo.toml` and a directory named
+# `target`.
+all_cache_directories=$(find /home -type d \( -name build -execdir test -f "x.py" \; -o -name target -execdir test -f "Cargo.toml" \; \) -print | sort | uniq)
```

Comment:
You may also want to sweep `build-rust-analyzer` too, which is a secondary build dir recommended for r-a workflows: https://rustc-dev-guide.rust-lang.org/building/suggested.html#configuring-rust-analyzer-for-rustc

Reply:
I'll make a note of this in #390, which tracks rewriting the script in Rust. I fear adding more conditions to the `find` command will get unmanageable quickly. 🙈